sdks/unity/AgonesSdkClient.cs

```diff
@@ -0,0 +1,179 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Newtonsoft.Json;
+using System.Collections;
+using System.Text;
+using UnityEngine;
+using UnityEngine.Networking;
+
+namespace Agones.SDK
+{
+    /// <summary>
+    /// JSON payload object
+    /// </summary>
+    internal class KeyValueMessage
+    {
+        public string key;
+        public string value;
+    }
+
+    /// <summary>
+    /// Agones SDK Unity Client
+    /// </summary>
+    public class AgonesSdkClient : MonoBehaviour
+    {
+        private const string emptyPayload = "{}";
+        private static byte[] emptyPayloadBytes;
+        private WaitForSeconds wait = new WaitForSeconds(0.5f);
+        private float updateInterval = 0.5f;
+
+        /// <summary>
+        /// Gets or sets the rate at which Health() should be called
+        /// </summary>
+        public float UpdateInterval
+        {
+            get
+            {
+                return this.updateInterval;
+            }
+            set
+            {
+                this.updateInterval = value;
+                this.wait = new WaitForSeconds(value);
+            }
+        }
+
+        /// <summary>
+        /// Unity Start
+        /// </summary>
+        private void Start()
+        {
+            AgonesSdkClient.emptyPayloadBytes = Encoding.UTF8.GetBytes(AgonesSdkClient.emptyPayload);
+
+            if (LaunchOptions.AgonesEnabled)
+            {
+                this.StartCoroutine(this.UpdateLoop());
+                MainContext.Instance.InstanceBinder.Bind<AgonesSdkClient>(this);
+            }
+        }
+
+        /// <summary>
+        /// This tells Agones that the Game Server is ready to take player connections.
+        /// Once a Game Server has specified that it is Ready, then the Kubernetes GameServer
+        /// record will be moved to the Ready state, and the details for its public address
+        /// and connection port will be populated.
+        /// </summary>
+        public void Ready()
+        {
+            AgonesSdkClient.EmptyPost("http://localhost:59358/ready");

```

Comment:
Is this port static among all usecases? The target URL should maybe have Editor-visible components.